types/node/index.d.ts

```diff
@@ -5219,6 +5220,21 @@ declare module "util" {
         __promisify__: TCustom;
     }
 
+    export function callbackify(fn: () => Promise<void>): (callback: (err: NodeJS.ErrnoException) => void) => void;
+    export function callbackify<TResult>(fn: () => Promise<TResult>): (callback: (err: NodeJS.ErrnoException, result: TResult) => void) => void;
+    export function callbackify<T1>(fn: (arg1: T1) => Promise<void>): (arg1: T1, callback: (err: NodeJS.ErrnoException) => void) => void;
```

Comment:
I don't think these `void` overloads are necessary; `void` could be just another `TResult`.

Reply:
Ping @alvis

Reply:
Sorry for the belated reply.

They are not overloads. In fact, they've different structure of return depending on whether `arg1` is passed to `fn`.

Reply:
Compare these two overloads:
```ts
export function callbackify(fn: () => Promise<void>): (callback: (err: NodeJS.ErrnoException) => void) => void;
export function callbackify<TResult>(fn: () => Promise<TResult>): (callback: (err: NodeJS.ErrnoException, result: TResult) => void) => void;
```
Isn't the first overload just the second overload with `void` in place of `TResult`? It's only missing `result: void`, but presumably that's what the second callback parameter would be.

Reply:
Got it. The reason of having another one is for the benefit of `result` in the callback function. Without the first declaration, typescript will suggest that the callback has two arguments instead of one, and the type of `result` is `{}`.